x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -688,7 +688,7 @@ private static void checkFullTextQueryFunctions(LogicalPlan plan, Set<Failure> f
                 plan,
                 condition,
                 Match.class,
-                lp -> (lp instanceof Limit == false),
+                lp -> (lp instanceof Limit == false) && (lp instanceof Aggregate == false),
```

Comment:
This makes sense to me - but in these types of situations I prefer to have an "allow list" of commands that can be used before `match`, rather than check for the ones that cannot be used.
This is what we do for `qstr` and IMO it's a bit safer, because we do not have to worry when we introduce a new command that we need to immediately update this check here. Worst case when we add a new command we have a limitation in place that we can lift any time.
The other way around, where we have a "deny" list like we do now, when we add a new command, the worst case is that the ES|QL query fails in a way we did not predict.

Reply:
That makes sense. As this is temporary until https://github.com/elastic/elasticsearch/issues/115311 is fixed, I can revisit the implementation when we change this again.